app/core/src/main/java/com/fsck/k9/mailstore/K9BackendFolder.kt

```diff
@@ -245,31 +246,41 @@ class K9BackendFolder(
     override fun getFolderExtraString(name: String): String? {
         return database.getStringOrNull(
                 table = "folder_extra_values",
-                column = "value_string"
+                column = "value_text",
+                selection = "name = ? AND folder_id = ?",
+                selectionArgs = *arrayOf(name, databaseId)
         )
     }
 
     override fun setFolderExtraString(name: String, value: String) {
-        database.setString(
-                table = "folder_extra_values",
-                column = "value_string",
-                value = value
-        )
+        database.execute(false) { db ->
+            val contentValues = ContentValues().apply {
+                put("name", name)
+                put("value_text", value)
+                put("folder_id", databaseId)
+            }
+            db.insertWithOnConflict("folder_extra_values", null, contentValues, SQLiteDatabase.CONFLICT_REPLACE)
+        }
     }
 
     override fun getFolderExtraNumber(name: String): Long? {
         return database.getLongOrNull(
                 table = "folder_extra_values",
-                column = "value_integer"
+                column = "value_integer",
+                selection = "name = ? AND folder_id = ?",
+                selectionArgs = *arrayOf(name, databaseId)
         )
     }
 
     override fun setFolderExtraNumber(name: String, value: Long) {
-        database.setLong(
```

Comment:
huh. did this just ignore the "name" parameter before and set this for all folders?

Reply:
It was broken in more ways than one :cry: 
I believe the old version created a SQL statement referencing an non-existent column.